crates/core/src/factories/core_publish/test.rs

```diff
@@ -0,0 +1,156 @@
+        op_store,
+        url.unwrap(),
+    )
+}
+
+#[tokio::test(flavor = "multi_thread")]
+async fn published_ops_can_be_retrieved() {
+    enable_tracing();
+
+    let (core_publish_1, op_store_1, _) =
+        setup_test(&CorePublishConfig::default()).await;
+    let (_core_publish2, op_store_2, url_2) =
+        setup_test(&CorePublishConfig::default()).await;
+
+    let incoming_op_1 = MemoryOp::new(Timestamp::now(), vec![1]);
+    let incoming_op_id_1 = incoming_op_1.compute_op_id();
+    let incoming_op_2 = MemoryOp::new(Timestamp::now(), vec![2]);
+    let incoming_op_id_2 = incoming_op_2.compute_op_id();
+
+    op_store_1
+        .process_incoming_ops(vec![incoming_op_1.into(), incoming_op_2.into()])
+        .await
+        .unwrap();
+
+    core_publish_1
+        .publish_ops(
+            vec![incoming_op_id_1.clone(), incoming_op_id_2.clone()],
+            url_2,
+        )
+        .await
+        .unwrap();
+
+    let ops = iter_check!(1000, {
+        let ops = op_store_2
+            .retrieve_ops(vec![
+                incoming_op_id_1.clone(),
+                incoming_op_id_2.clone(),
+            ])
+            .await
+            .unwrap();
+
+        if ops.len() == 2 {
+            return ops;
+        }
+    });
+
+    assert!(ops.len() == 2);
+}
+
+#[tokio::test(flavor = "multi_thread")]
+async fn publish_to_invalid_url_does_not_impede_subsequent_publishes() {
+    enable_tracing();
+
+    let (core_publish_1, op_store_1, _) =
+        setup_test(&CorePublishConfig::default()).await;
+    let (_core_publish2, op_store_2, url_2) =
+        setup_test(&CorePublishConfig::default()).await;
+
+    let incoming_op_1 = MemoryOp::new(Timestamp::now(), vec![1]);
+    let incoming_op_id_1 = incoming_op_1.compute_op_id();
+    let incoming_op_2 = MemoryOp::new(Timestamp::now(), vec![2]);
+    let incoming_op_id_2 = incoming_op_2.compute_op_id();
+
+    op_store_1
+        .process_incoming_ops(vec![incoming_op_1.into(), incoming_op_2.into()])
+        .await
+        .unwrap();
+
+    let op_ids = vec![incoming_op_id_1.clone(), incoming_op_id_2.clone()];
+
+    // Publish to a non-existing Url
+    core_publish_1
+        .publish_ops(
+            op_ids.clone(),
+            Url::from_str("ws://notanexistingurl:80").unwrap(),
+        )
+        .await
+        .unwrap();
+
+    // Publish to an existing Url to verify that the prior publishing
+    // to a non-existing Url did not cause an error that would
+    // break the publishing flow
+    core_publish_1.publish_ops(op_ids, url_2).await.unwrap();
+
+    let ops = iter_check!(1000, {
+        let ops = op_store_2
+            .retrieve_ops(vec![
+                incoming_op_id_1.clone(),
+                incoming_op_id_2.clone(),
+            ])
+            .await
+            .unwrap();
+
+        if ops.len() == 2 {
+            return ops;
+        }
+    });
+
+    assert!(ops.len() == 2);
```

Comment:
Yes, that looks like a good test to me.
```suggestion
    iter_check!(1000, {
        let ops = op_store_2
            .retrieve_ops(vec![
                incoming_op_id_1.clone(),
                incoming_op_id_2.clone(),
            ])
            .await
            .unwrap();

        if ops.len() == 2 {
            break;
        }
    });
```

Reply:
@ThetaSinner and I were discussing that the final assert is not needed really (when writing the other test which has the same logic in that regard) but we concluded that it would be good to let it there anyway for the purpose of readability since it's harder to understand that this check is already done implicitly by breaking the loop...

Reply:
I still prefer that. I found too many tests in Holochain that use macros to wait for something but the final condition was slightly off and it took a long time to decode, a really clear assertion at the end of the test makes that hard to get wrong. I could live with it being removed in this case but in general, I think it's a helpful pattern

Reply:
Maybe we can rename the macro to something that implies a condition to break the loop, so that assertion is clear. This iter_check pattern is used in the majority of places in this repo.